genesyscloud/resource_genesyscloud_journey_segment_test.go

```diff
@@ -0,0 +1,104 @@
+package genesyscloud
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/mypurecloud/platform-client-sdk-go/v72/platformclientv2"
+)
+
+func TestAccResourceJourneySegmentSession(t *testing.T) {
+	runTestCase(t, "basic_session_attributes")
```

Comment:
This fails for me with `API Error: 400 - Event type was not found with name: EventName`

Reply:
I'm running against a prod org in us-east-1

Reply:
Interesting, why does it not fail at DCA?

Based on the API documentation it is not a required parameter and expected to accept any value.
```
journey
(Journey) The pattern of rules defining the segment.
patterns
(JourneyPattern[]) A list of zero or more patterns to match.
criteria
(Criteria[]) A list of one or more criteria to satisfy.
count
(integer) The number of times the pattern must match.
streamType
(string required) The stream type for which this pattern can be matched on.Valid values: Web, Custom, Conversation.
sessionType
(string required) The session type for which this pattern can be matched on.
eventName
(string) The name of the event for which this pattern can be matched on.
```
We have no extra information for this field and UI also does not make to set it. Maybe we should just remove from our tes cases.

Reply:
I removed the event_name from test configs. Based on the code, if I provide one  than it should exist. Probably we had luck in DCA...
https://bitbucket.org/inindca/journey-rules-engine/src/master/src/main/java/com/genesys/journey/rulesengine/service/PatternValidator.java?at=master#lines-122